spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java

```diff
@@ -466,30 +472,33 @@ static void queryUsingIndex(DatabaseClient dbClient) {
             .bind("EndTitle")
             .to("Goo")
             .build();
-
-    ResultSet resultSet = dbClient.singleUse().executeQuery(statement);
-    while (resultSet.next()) {
-      System.out.printf(
-          "%d %s %s\n",
-          resultSet.getLong("AlbumId"),
-          resultSet.getString("AlbumTitle"),
-          resultSet.isNull("MarketingBudget") ? "NULL" : resultSet.getLong("MarketingBudget"));
+    // We use a try-with-resource block to automatically release resources held by ResultSet.
+    try (ResultSet resultSet = dbClient.singleUse().executeQuery(statement)) {
+      while (resultSet.next()) {
+        System.out.printf(
+            "%d %s %s\n",
+            resultSet.getLong("AlbumId"),
+            resultSet.getString("AlbumTitle"),
+            resultSet.isNull("MarketingBudget") ? "NULL" : resultSet.getLong("MarketingBudget"));
+      }
     }
   }
   // [END spanner_query_data_with_index]
 
   // [START spanner_read_data_with_index]
   static void readUsingIndex(DatabaseClient dbClient) {
-    ResultSet resultSet =
+    // We use a try-with-resource block to automatically release resources held by ResultSet.
```

Comment:
try-with-resources isn't that novel these days.

Reply:
Opting to leave these comments in for now.